ui/StatusQ/src/wallet/managetokensmodel.cpp

```diff
@@ -0,0 +1,192 @@
+}
+
+QHash<int, QByteArray> ManageTokensModel::roleNames() const
+{
+    static const QHash<int, QByteArray> roles {
+        {SymbolRole, kSymbolRoleName},
+        {NameRole, kNameRoleName},
+        {CommunityIdRole, kCommunityIdRoleName},
+        {CommunityNameRole, kCommunityNameRoleName},
+        {CommunityImageRole, kCommunityImageRoleName},
+        {CollectionUidRole, kCollectionUidRoleName},
+        {CollectionNameRole, kCollectionNameRoleName},
+        {BalanceRole, kEnabledNetworkBalanceRoleName},
+        {CurrencyBalanceRole, kEnabledNetworkCurrencyBalanceRoleName},
+        {CustomSortOrderNoRole, kCustomSortOrderNoRoleName},
+        {TokenImageRole, kTokenImageRoleName},
+    };
+
+    return roles;
+}
+
+QVariant ManageTokensModel::data(const QModelIndex& index, int role) const
+{
+    if (!checkIndex(index, QAbstractItemModel::CheckIndexOption::IndexIsValid | QAbstractItemModel::CheckIndexOption::ParentIsInvalid))
+        return {};
+
+    const auto& token = m_data.at(index.row());
+
+    switch(static_cast<TokenDataRoles>(role))
+    {
+    case SymbolRole: return token.symbol;
+    case NameRole: return token.name;
+    case CommunityIdRole: return token.communityId;
+    case CommunityNameRole: return token.communityName;
+    case CommunityImageRole: return token.communityImage;
+    case CollectionUidRole: return token.collectionUid;
+    case CollectionNameRole: return token.collectionName;
+    case BalanceRole: return token.balance;
+    case CurrencyBalanceRole: return token.currencyBalance;
+    case CustomSortOrderNoRole: return token.customSortOrderNo;
+    case TokenImageRole: return token.image;
+    }
+
+    return {};
+}
+
+bool ManageTokensModel::dirty() const
+{
+    return m_dirty;
+}
+
+void ManageTokensModel::setDirty(bool flag)
+{
+    if (m_dirty == flag) return;
+    m_dirty = flag;
+    emit dirtyChanged();
+}
+
+void ManageTokensModel::saveCustomSortOrder()
+{
+    const auto count = rowCount();
+    for (auto i = 0; i < count; i++) {
+        TokenData newToken{m_data.at(i)};
+        if (newToken.communityId.isEmpty()) {
+            newToken.customSortOrderNo = i;
+        } else {
+            const auto communityIdx = m_communityIds.indexOf(newToken.communityId) + 1;
+            newToken.customSortOrderNo = i + (communityIdx * 100'000);
+        }
+        m_data[i] = newToken;
+    }
+    emit dataChanged(index(0, 0), index(count - 1, 0), {TokenDataRoles::CustomSortOrderNoRole});
+}
+
+void ManageTokensModel::applySort()
+{
+    emit layoutAboutToBeChanged({}, QAbstractItemModel::VerticalSortHint);
+
+    // clazy:exclude=clazy-detaching-member
+    std::stable_sort(m_data.begin(), m_data.end(), [this](const TokenData& lhs, const TokenData& rhs) {
+        return lhs.customSortOrderNo < rhs.customSortOrderNo;
+    });
+
+    emit layoutChanged({}, QAbstractItemModel::VerticalSortHint);
+}
+
+bool ManageTokensModel::hasCommunityIdToken(const QString& communityId) const
```

Comment:
I can do, yeah

Reply:
Ah right... no because: QML doesn't support overloaded method calls, I'll need (for the collectibles) to keep a distinction between a community group and collection group

Reply:
Ok.. then maybe `containsCommunityToken` and `containsCollectionToken`??

Reply:
Hmm, I'll have to see how that fits the Collectibles; afaik they can contain both a "collection" and "communityId" :) But the naming would change the semantics